packages/components/time-picker/src/time-picker-com/basic-time-spinner.vue

```diff
@@ -232,12 +232,36 @@ const scrollDown = (step: number) => {
   let now = unref(timePartials)[label]
   const total = currentScrollbar.value === 'hours' ? 24 : 60
   now = (now + step + total) % total
-
+  now = findNextUnDisabled(label, now, step, total)
   modifyDateField(label, now)
   adjustSpinner(label, now)
   nextTick(() => emitSelectRange(label))
 }
-
+const findNextUnDisabled = (
+  type: TimeUnit,
+  value: number,
+  step: number,
+  total: number
+) => {
+  let res = value
+  const list = unref(timeList)[type]
+  let isDisabled: boolean = list[value]
+  if (isDisabled) {
+    const len: number = list.length
+    while (isDisabled) {
```

Comment:
The judgment conditions are not comprehensive enough. When all options are not selectable, an infinite loop will be entered.

Reply:
This infinite loop problem still exists

Reply:
> This infinite loop problem still exists
![image](https://user-images.githubusercontent.com/32354856/181919401-fa88bebe-4815-4dfa-b7b3-5ac9963af9ba.png)
When all options are disabled, I use set to make a judgment and return directly. Are there other scenarios that will lead to infinite loops?

Reply:
Looks like you are overcomplicating things.

Maybe adjust the parameter names to make it easier to understand

```
value -> now
res -> next
```

They are both indexes of the list array, and when they are equal again, it means that all options have been traversed.